lib/active_model/serializer/reflection.rb

```diff
@@ -54,7 +61,7 @@ def build_association(subject, parent_serializer_options)
             reflection_options[:virtual_value] = association_value.try(:as_json) || association_value
           end
         elsif !association_value.nil? && !association_value.instance_of?(Object)
-          reflection_options[:virtual_value] = association_value
+          reflection_options[:virtual_value] = association_value.try(:as_json) || association_value
```

Comment:
can you write some comments about why try is being used and a quick blurb about when association_value might not have `as_json`

thanks ;-)